### PR TITLE
[13.0][IMP] project_stock: Confirm the new stock movements in the "Check availability materials" button

### PR DESCRIPTION
**Pull request by victoralmau**
Confirm the new stock movements in the "**Check availability materials**" button (if movements have been added after confirming the materials).

Please @pedrobaeza and @sergio-teruel can you review it?

@Tecnativa TT35602

### CODE REVIEW

File: project_stock/models/project_task.py [hunk 1]
```diff
@@ -122,6 +122,9 @@ def action_confirm(self):
         self.action_assign()
 
     def action_assign(self):
+        self.move_ids.filtered(lambda x: x.state == "draft").write(
```
Comment (pedrobaeza) [inline project_stock/models/project_task.py]: Why not calling the method to confirm instead?

File: project_stock/models/project_task.py [hunk 2]
```diff
@@ -118,10 +118,11 @@ def _prepare_procurement_group_vals(self, values):
         return {"name": values["name"]}
 
     def action_confirm(self):
-        self.mapped("move_ids").write({"state": "confirmed"})
+        self.mapped("move_ids")._action_confirm()
         self.action_assign()
 
     def action_assign(self):
+        self.move_ids.filtered(lambda x: x.state == "draft")._action_confirm()
```
Comment (pedrobaeza) [inline project_stock/models/project_task.py]: Call `self.action_confirm()`

File: project_stock/models/project_task.py [hunk 3]
```diff
@@ -118,11 +118,11 @@ def _prepare_procurement_group_vals(self, values):
         return {"name": values["name"]}
 
     def action_confirm(self):
-        self.mapped("move_ids").write({"state": "confirmed"})
-        self.action_assign()
+        self.mapped("move_ids")._action_confirm()
+        self.mapped("move_ids")._action_assign()
```
Comment (pedrobaeza) [inline project_stock/models/project_task.py:125]: Why is this here?

Comment (victoralmau) [reply project_stock/models/project_task.py:125]: This change is related to https://github.com/OCA/project/pull/914#discussion_r854118954.

I think it's good to explain the process to understand everything.

**Before:**
- "Confirm Materials" button confirmed stock moves with the `action_confirm()` method.
- "Check availability" button called the `_action_assign()` method to reserve the stock.
- Additionally, "Confirm materials" button called the method of the "Check availability" button (to avoid clicking on confirm and always check availability).

**Actual use case:**
- The task is changed to "In Progress" stage.
- Stock moves are added and confirmed with "Confirm materials" button.
- The task is changed to "Completed" stage.
- The task is changed back to "In Progress" stage.
- Some extra stock moves is added.
- "Check availability" button does nothing for the new line because the status of line is draft (not confirmed).

**Changes done:**
- [x] Changed `self.mapped("move_ids").write({"state": "confirmed"})` to `self.mapped("move_ids")._action_confirm()`.
- [x] Changed `self.action_assign()` to `self.mapped("move_ids")._action_assign()`.
- [x] Changed the `action_assign()` event to call `action_confirm()` event and solve the previous use case.

Comment (pedrobaeza) [reply project_stock/models/project_task.py:125]: My comment was not asking for that. There's no sense to have 2 methods that do the same

File: project_stock/models/project_task.py [hunk 4]
```diff
@@ -118,10 +118,7 @@ def _prepare_procurement_group_vals(self, values):
         return {"name": values["name"]}
 
     def action_confirm(self):
-        self.mapped("move_ids").write({"state": "confirmed"})
-        self.action_assign()
-
-    def action_assign(self):
+        self.mapped("move_ids")._action_confirm()
         self.mapped("move_ids")._action_assign()
```
Comment (pedrobaeza) [inline project_stock/models/project_task.py]: It seems I'm not able to transfer you what I mean. Let me put the code:

```suggestion
    def action_confirm(self):
        self.mapped("move_ids")._action_confirm()

    def action_assign(self):
        self.action_confirm()
        self.mapped("move_ids")._action_assign()
```

**Review by pedrobaeza — APPROVED** [review]

**Review by sergio-teruel — APPROVED** [review]
